ES6

bkendall opened the pull request:
Rewrote the module in ES6. When published, it builds the source to a common js module so it is still usable by `0.10`.

Commit:
:open_hands: es6

Commit:
move tests to src

Commit:
prepublish script, remove coverage, ignore file

Commit:
ignore log

Commit:
I have seen the light of bluebird...

Commit:
remove log

Commit:
test on node 0.10

Commit:
protect npm build from unnecessary runs

Commit:
ignore hidden files

bkendall committed:
remove all NODE_PATH related things

bkendall committed:
remove extra comment

Commit:
disable email notifications

Commit:
always use parens with new standard

Commit:
fix coverage and reporting

Commit:
adding coverage badge back

Commit:
istanbul + isparta -> nyc

Commit:
make html and text reports

Commit:
add test for defaulting name

Commit:
100% threshold

Commit:
bump npm to 2 if at 1

Commit:
remove istanbul

Commit:
deploy from travis on tags on node 4

anandkumarpatel commented on src/experiment.js:
```diff
@@ -0,0 +1,251 @@
+        .then((_) => {
+          let promises = []
+
+          const shuffle = knuth_shuffle.knuthShuffle
```
cupid shuffle :)

bkendall replied:
uhh... okay?

bkendall replied:
https://www.youtube.com/watch?v=h24_zoqu4_Q

anandkumarpatel commented on src/observation.js:
```diff
@@ -0,0 +1,78 @@
+  static create (name, experiment, fn) {
+    const observation = new Observation(name, experiment, fn)
+    return Promise.resolve()
+      .then((_) => (observation.fn()))
```
why the `_` ? I don't think its needed

bkendall replied:
interesting you bring this up. this is an artifact of how I wrote this and then upgraded `standard`. [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) says functions with no arguments need either `()` or `_` before the arrow. Until recently, `standard` didn't care, but now they want empty parens rather than an underscore. I removed them just now, but that's the logic

Commit:
remove underscores

anandkumarpatel commented on src/experiment.js:
```diff
@@ -0,0 +1,251 @@
+   * @return {Object} Cleaned (if applicable) value.
+   */
+  clean_value (value) {
+    debug('clean_value')
```
would you want to put the value in the debug statement?

bkendall replied:
not _really_... the reason I have all those `debug` statements was to track the flow of things for a while. they can be expanded in the future, but not interested in doing that now

Commit:
set duration to be above 4

the test delays for 5, but sometimes CI runs fast enough (and locally) that the delay is exactly 5 as well. let's just bring it down a har

anandkumarpatel commented:
+1 look cool

Commit:
add nyc output
